api/src/main/java/jakarta/servlet/http/HttpServletRequest.java

```diff
@@ -288,11 +288,15 @@ public String toString() {
      *
      * @implSpec The default implementation returns null.
      *
-     * @return a {@link PushBuilder} for issuing server push responses from the current request, or null if push is not
-     * supported
+     * @return a {@link PushBuilder} for issuing server push responses from the current request, or {@code null} if push is
+     * not supported. Note that some implementations may opt not to support server push and will therefore always return
+     * {@code null}
      *
      * @since Servlet 4.0
+     *
+     * @deprecated Use early hints via TBD instead
```

Comment:
Not a huge fan of the TBD.   
```suggestion
     * @deprecated In favor of 103 early hints
```

When we have an early hint mechanism, then we can remember to update this just as well as remembering to remove the TBD

Reply:
Fair enough. I'll update the PR.